### ISSUE TITLE
Add homescreen benchmark

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#21112]
_tldr; make it easy to run & demonstrate capability with a real benchmark test_

We want to understand if the changes to the homescreen are having an impact on performance. Let's add a new benchmark **to be run locally** that isolates the homescreen. Here is a possible breakdown of work:
- [ ] Figure out what framework to use for non-start up, UI benchmarking. The default option is Jetpack Benchmark. Currently, it has limitations (can only benchmark in tight loops, it might only run on locally built builds) but presumably those will go away in time and it is a natural way to interact with Android. Note we'll need to pre-populate data into the app too
- [ ] Validate that the benchmark can be used to detect regressions
- [ ] Set up the framework, making it easy to run & get results. For Jetpack Benchmark, we set it up in https://github.com/mozilla-mobile/fenix/issues/19901 but it doesn't return the results in an intuitive way nor did it run well from Android Studio. Perhaps we need a wrapper script?
- [ ] (optional) Write a simple version of the test
- [ ] Write a more-advanced version of the test that pre-populates data: some new functionality won't be shown until the user has some data so we should make sure we test those code paths

**Comment by mcomella** [comment]
This was broken into separate bugs, e.g. https://github.com/mozilla-mobile/fenix/issues/21699 & https://github.com/mozilla-mobile/fenix/issues/21700
